hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -155,6 +160,8 @@ public abstract class HoodieBackedTableMetadataWriter<I> implements HoodieTableM
   protected StorageConfiguration<?> storageConf;
   protected final transient HoodieEngineContext engineContext;
   protected final List<MetadataPartitionType> enabledPartitionTypes;
+  protected final MetadataWriteHandler metadataWriteHandler;
```

Comment:
can you add java docs on the purpose of write handler

Reply:
I feel like maybe we just sub-class the whole `HoodieBackedTableMetadataWriter` instead of introducing `MetadataWriteHandler` on existing code, this makes the code clean and more easier to remove the legacys in the future.

Reply:
yeah, we also wanted to do that. 
but due to our hierarchy, we could not come up w/ an elegant way to do that. 

bcoz, we have HoodieBackedTableMetadatWriter. and Flink, Java and SparkBackedTableMetadatWriter extends from this. 

So, not easy to make entire class HoodieBackedTableMetadatWriter as V1 and V2. So, we went w/ introducing a new write handler internally within HoodieBackedTableMetadatWriter. 

Let us know if you have better ideas.

Reply:
We can add a base extended class from `HoodieBackedTableMetadatWriter`, then share it with different engines.